portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,19 +27,44 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> names;
-
-  @Override
-  public void init() {
-      names = new ArrayList<>(Arrays.asList("Harry", "Ava", "Mia"));
-  }
-
+  private List<String> comments = new ArrayList<String>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Gson gson = new Gson();
-    String namesJson = gson.toJson(names);
+    String commentsJson = gson.toJson(comments);
 
     response.setContentType("application/json;");
-    response.getWriter().println(namesJson);
+    response.getWriter().println(commentsJson);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the comment from the form
+    String newComment = getNewComment(request);
+    if (newComment == "") {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter a valid comment.");
+      return;
+    }
+    
+    comments.add(newComment);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  private String getNewComment(HttpServletRequest request) {
+    String newComment = request.getParameter("new-comment");
+
+    byte[] commentBytes = null;
+    try {
+      commentBytes = newComment.getBytes("UTF-8");
+    } catch (UnsupportedEncodingException e) {
+      System.err.println("Invalid comment: " + newComment);
```

Comment:
It might be good to raise an exception here, then catch it in `doPost` and send an appropriate error back to the browser (e.g. a HTTP 5xx error).  We can do similar for the "" scenario too.

This way the browser can show an error to the user.  Let me know if you need some more details here!

Reply:
I changed it, please tell me if there is a better way to send an error to the browser.

Reply:
This is a good start.  When there is a problem and you throw ServletException, the browser will send a HTTP 500 error back to the browser.  500 means just a "internal server error".

This is useful ... but is this really an error in the server, or an error with the data the user sent us? We could provide a better error code to the browser such as HTTP 400 which is a "Bad Request" (i.e. the user sent us some bad info) that might help end users understand what they did wrong, e.g.

```
try {
  newComment = getNewComment(request);
} catch (UnsupportedEncodingException e) {
  // Send a HTTP 400 Bad Request response if the user provided bad data.
  response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
} catch (Exception e) {
  // If there was some other exception that we didn't already know what to do with
  // just send a general HTTP 500 error.
  response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getmessage());
}
```

Then your javascript code can look at the HTTP response code and do something appropriate to e.g. ask the user to enter valid info, or an error message if there was a server error (I don't think your javascript does anything with errors at the moment).

Don't worry about implementing it for this PR - just something to think about for the future! :-)